src/backend/distributed/deparser/deparse_function_stmts.c

```diff
@@ -268,31 +268,128 @@ static void
 AppendDefElemSet(StringInfo buf, DefElem *def)
 {
 	VariableSetStmt *setStmt = castNode(VariableSetStmt, def->arg);
-	char *setVariableArgs = ExtractSetVariableArgs(setStmt);
 
 	switch (setStmt->kind)
 	{
 		case VAR_SET_VALUE:
 		{
-			appendStringInfo(buf, " SET %s = %s", setStmt->name, setVariableArgs);
+			ListCell *varArgCell = NULL;
+			ListCell *firstCell = list_head(setStmt->args);
+
+			/* see flatten_set_variable_args in postgres's utils/misc/guc.c */
```

Comment:
This should probably live in its own function with a comment to clarify the quoting difference with flatten_set_variable_args.